flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/JobGraph.java

```diff
@@ -584,17 +584,25 @@ public void uploadUserArtifacts(InetSocketAddress blobServerAddress, Configurati
 		if (!userArtifacts.isEmpty()) {
 			try (BlobClient blobClient = new BlobClient(blobServerAddress, clientConfig)) {
 				for (Map.Entry<String, DistributedCache.DistributedCacheEntry> userArtifact : userArtifacts.entrySet()) {
-
-					final PermanentBlobKey key = blobClient.uploadFile(jobID,
-						new Path(userArtifact.getValue().filePath));
-
-					DistributedCache.writeFileInfoToConfig(
-						userArtifact.getKey(),
-						new DistributedCache.DistributedCacheEntry(
-							userArtifact.getValue().filePath,
-							userArtifact.getValue().isExecutable,
-							InstantiationUtil.serializeObject(key)),
-						jobConfiguration);
+					Path filePath = new Path(userArtifact.getValue().filePath);
+
+					if (filePath.getFileSystem().isDistributedFS()) {
```

Comment:
The file-system class may not be accessible on the client, so you have to guard this with a try-catch block and write it into the config in both cases.